tests/router_test.ts

```diff
@@ -1500,6 +1500,67 @@ scenarios.forEach(function(scenario) {
       });
   });
 
+  test('abort query param only', function(assert) {
+    assert.expect(6);
+    map(assert, function(match) {
+      match('/').to('index');
+    });
+
+    routes = {
+      search: createHandler('search'),
+    };
+
+    let newParam = false;
+    let initial = true;
+
+    router.updateURL = function(updateUrl) {
+      url = updateUrl;
+      if (!initial) {
+        assert.ok(false, 'updateURL should not be called');
+      }
+    };
+
+    router.routeWillChange = (transition: Transition) => {
+      if (!transition.isAborted) {
+        if (newParam) {
+          assert.deepEqual(transition.to!.queryParams, { term: 'b' }, 'going to page with qps');
+          assert.deepEqual(
+            isPresent(transition.from) && transition.from!.queryParams,
+            {},
+            'from never has qps'
+          );
+        } else {
+          assert.equal(transition.from, null);
```

Comment:
Can we `strictEqual` instead (since `equal` will coerce IIRC)?

```suggestion
          assert.strictEqual(transition.from, null);
```